resources/src/operation/create_account.rs

```diff
@@ -0,0 +1,70 @@
+use amount::Amount;
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct CreateAccountFields {
+    pub account: String,
+    pub funder: String,
+    pub starting_balance: Amount,
+}
+
+impl CreateAccountFields {
+    pub fn new(
+        account: Option<String>,
+        funder: Option<String>,
+        starting_balance: Option<Amount>,
+    ) -> Result<CreateAccountFields, String> {
+        if account.is_none() || funder.is_none() || starting_balance.is_none() {
+            Err("Account, funder and starting balance required.".to_string())
```

Comment:
This error doesn't look tested.

Also, if they can never be none, why even allow it in the type? If we make `new` a public function, we probably don't want to insert required optionals.

Reply:
They are options because at the moment of deserialization we don't know what type they will be.  That means all the fields in intermediate_operation must be options.  Those options then must be validated and unwrapped for each error type.  It feels better organized to put this logic in a new method separated into each OpDetail's mod rather than making the Operation deserializer handle all that logic.

Reply:
I think that `new` should constitute creating a new instance of the struct. Passing deserialization to the new command as a `pub` function is creating complexity that doesn't need to be there. It should be captured in the deserialization process itself.